pages/api/stats/index.ts

```diff
@@ -1,25 +1,47 @@
 import { ObjectId } from 'bson';
-import mongoose from 'mongoose';
+import mongoose, { SaveOptions } from 'mongoose';
 import type { NextApiResponse } from 'next';
+import AchievementType from '../../../constants/achievementType';
 import Discord from '../../../constants/discord';
 import { ValidArray, ValidObjectId, ValidType } from '../../../helpers/apiWrapper';
 import queueDiscordWebhook from '../../../helpers/discordWebhook';
 import { TimerUtil } from '../../../helpers/getTs';
 import { logger } from '../../../helpers/logger';
-import { createNewRecordOnALevelYouBeatNotifications } from '../../../helpers/notificationHelper';
+import { createNewAchievementNotification, createNewRecordOnALevelYouBeatNotifications } from '../../../helpers/notificationHelper';
 import revalidateLevel from '../../../helpers/revalidateLevel';
 import validateSolution from '../../../helpers/validateSolution';
 import dbConnect from '../../../lib/dbConnect';
 import withAuth, { NextApiRequestWithAuth } from '../../../lib/withAuth';
 import Level from '../../../models/db/level';
 import Record from '../../../models/db/record';
 import Stat from '../../../models/db/stat';
-import { LevelModel, PlayAttemptModel, RecordModel, StatModel, UserModel } from '../../../models/mongoose';
+import User from '../../../models/db/user';
+import { AchievementModel, LevelModel, PlayAttemptModel, RecordModel, StatModel, UserModel } from '../../../models/mongoose';
 import { AttemptContext } from '../../../models/schemas/playAttemptSchema';
 import { queueCalcPlayAttempts, queueRefreshIndexCalcs } from '../internal-jobs/worker';
 import { MatchMarkCompleteLevel } from '../match/[matchId]';
 import { forceCompleteLatestPlayAttempt } from '../play-attempt';
 
+// called when we have confirmed the user has beaten the level, but user score hasn't been updated yet
+function issueAchievements(user: User, options: SaveOptions) {
+  const promises = [];
+
+  if (user.score === 99) {
```

Comment:
Need to make this idempotent (>= 99) so that users already over 100 will get the achievement.